### ISSUE TITLE
Transformation with incorrect z_in units

### ISSUE THREAD

**Issue by jmilloy** [pyproj4/pyproj#688]
#### Code Sample

```
>>> import pyproj
>>> a = pyproj.Proj('+proj=merc +vunits=cm')
>>> b = pyproj.Proj('+proj=longlat +datum=WGS84 +vunits=cm')
>>> transformer = pyproj.Transformer.from_proj(a, b, always_xy=True)
>>> transformer.transform(0, 0, 10)
(0.0, 0.0, 1000.0)
```

#### Problem description

The vertical height units are both centimeters, but they are incorrectly converted by the transformer.

Note that the transformer operation has the incorrect `z_in` units:

```
>>> transformer.operations[1]
<Coordinate Operation: Other Coordinate Operation>
Name: Transformation from unknown to unknown + Transformation from unknown to unknown (ballpark vertical transformation, without ellipsoid height to vertical height correction)
Method: PROJ-based operation method (approximate): +proj=unitconvert +z_in=m +z_out=cm
Area of Use:
- name: World
- bounds: (-180.0, -90.0, 180.0, 90.0)
```

Basically, `+proj=merc +vunits=cm` seems to be misinterpreted by the transformer, even though the height axis seem to be correct:

```
>>> pyproj.Proj('+proj=merc +vunits=cm')
Proj('+proj=merc +lon_0=0 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +vunits=cm +no_defs', preserve_units=True)

>>> pyproj.Proj('+proj=merc +vunits=cm').crs
<Compound CRS: +proj=merc +vunits=cm +type=crs>
Name: unknown
Axis Info [cartesian|vertical]:
- E[east]: Easting (metre)
- N[north]: Northing (metre)
- H[up]: Gravity-related height (centimetre)
Area of Use:
- undefined
Datum: World Geodetic System 1984
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich
Sub CRS:
- unknown
- unknown
```

#### Expected Output

```
(0.0, 0.0, 10.0)
```

#### Environment Information

```
$ python -m pyproj -v
pyproj info:
    pyproj: 2.6.1.post1
      PROJ: 7.0.1
  data dir: /home/jmilloy/Creare/Pipeline/_podpac38_/lib/python3.8/site-packages/pyproj/proj_dir/share/proj

System:
    python: 3.8.3 (default, May 14 2020, 20:11:43)  [GCC 7.5.0]
executable: /home/jmilloy/Creare/Pipeline/_podpac38_/bin/python
   machine: Linux-5.4.0-42-generic-x86_64-with-glibc2.27

Python deps:
       pip: 20.1.1
setuptools: 49.1.0
    Cython: None
(_podpac38_) jmilloy@o
```

#### Installation method
 - pip

**Comment by jmilloy** [comment]
I don't know if this is an issue in pyproj or in PROJ. How can I test that?

**Comment by snowman2** [comment]
Looks like a PROJ issue:
```
echo '0 0 10' | cs2cs -d 1 +proj=merc +vunits=cm +type=crs +to +proj=longlat +datum=WGS84 +vunits=cm +type=crs
0.0	0.0 1000.0
```

**Comment by jmilloy** [comment]
Thanks. I tested the same on my machine, but using `cs2cs` works as expected:

```
$ echo '0 0 10' | cs2cs -d 1 +proj=merc +vunits=cm +type=crs +to +proj=longlat +datum=WGS84 +vunits=cm +type=crs
0dE	0dN 10.000
```

Any idea why my cs2cs is working as expected, but my pyproj and your cs2cs share this bug? I can't really justify submitting a PROJ issue myself if I can't produce it.

Here is the verbose output:

```
$ cs2cs -d 1 +proj=merc +vunits=m +type=crs +to +proj=longlat +datum=WGS84 +vunits=cm +type=crs -v
# ---- From Coordinate System ----
#Mercator
#	Cyl, Sph&Ell
#	lat_ts=
# +proj=merc +vunits=m +ellps=WGS84
#--- following specified but NOT used
# +type=crs
# ---- To Coordinate System ----
#Lat/long (Geodetic alias)
#	
# +proj=longlat +datum=WGS84 +vunits=cm +ellps=WGS84 +towgs84=0,0,0
#--- following specified but NOT used
# +type=crs
```

It looks like my command line proj version might be different from the one pyproj is using (`4.9.3` vs `7.0.1`). That could be an explanation, but I don't see another proj binary on my machine anywhere.

```
$ proj
Rel. 4.9.3, 15 August 2016
```

**Comment by snowman2** [comment]
If you installed with wheels, then the new PROJ binaries weren't added. I would recommend raising this issue upstream: https://github.com/osgeo/proj

**Comment by jmilloy** [comment]
I understand why it should be reported upstream, but I don't know what to report because I can't produce the error in PROJ.

**Comment by snowman2** [comment]
You can use the output from my version:
```
$ cs2cs
Rel. 7.1.0, August 1st, 2020
usage: cs2cs [-dDeEfIlrstvwW [args]] [+opt[=arg] ...]
                   [+to +opt[=arg] ...] [file ...]
```
```
echo '0 0 10' | cs2cs -d 1 +proj=merc +vunits=cm +type=crs +to +proj=longlat +datum=WGS84 +vunits=cm +type=crs
0.0	0.0 1000.0
```

**Comment by jmilloy** [comment]
This is fixed in PROJ 7.1
